Sources/AsyncHTTPClient/ConnectionPool.swift

```diff
@@ -440,12 +453,31 @@ class HTTP1ConnectionProvider {
 
     private func makeChannel(preference: HTTPClient.EventLoopPreference,
                              logger: Logger) -> EventLoopFuture<Channel> {
-        return NIOClientTCPBootstrap.makeHTTP1Channel(destination: self.key,
-                                                      eventLoop: self.eventLoop,
-                                                      configuration: self.configuration,
-                                                      sslContextCache: self.pool.sslContextCache,
-                                                      preference: preference,
-                                                      logger: logger)
+        let connectionID = HTTPConnectionPool.Connection.ID.globalGenerator.next()
+        let eventLoop = preference.bestEventLoop ?? self.eventLoop
+        let deadline = NIODeadline.now() + (self.configuration.timeout.connect ?? .seconds(10))
```

Comment:
This hardcoded default is a bit wild. I'd rather we didn't put it here but instead modified the behaviour of the timeout structure.

Reply:
Fixed!